TractorMod/Framework/Attachments/AxeAttachment.cs

```diff
@@ -66,29 +81,37 @@ public override bool Apply(Vector2 tile, SObject tileObj, TerrainFeature tileFea
         {
             // clear twigs & weeds
             if (tileObj?.Name == "Twig" || tileObj?.Name.ToLower().Contains("weed") == true)
-                return this.UseToolOnTile(tool, tile);
+                return this.ClearDebris && this.UseToolOnTile(tool, tile);
 
             // cut non-fruit tree
             if (tileFeature is Tree)
-                return this.CutTrees && this.UseToolOnTile(tool, tile);
+            {
+                if ((tileFeature as Tree).tapped)
+                    return this.CutTappedTrees && this.UseToolOnTile(tool, tile);
+                else
+                    return this.CutTrees && this.UseToolOnTile(tool, tile);
+            }
 
             // cut fruit tree
             if (tileFeature is FruitTree)
                 return this.CutFruitTrees && this.UseToolOnTile(tool, tile);
 
-            // clear crops
-            if (tileFeature is HoeDirt dirt && dirt.crop != null && (dirt.crop.dead || this.ClearCrops))
-                return this.UseToolOnTile(tool, tile);
+            // clear dead crops
+            if (tileFeature is HoeDirt dirt && dirt.crop != null)
```

Comment:
The brackets are missing here, so only the immediate next block is inside the outer `if`. The new code is equivalent to this:
```cs
if (tileFeature is HoeDirt dirt && dirt.crop != null)
{
   if (dirt.crop.dead && this.ClearDeadCrops)
      return this.UseToolOnTile(tool, tile);
}
else if (!dirt.crop.dead && this.ClearLiveCrops)
   return this.UseToolOnTile(tool, tile);
```

To avoid confusion, I add brackets anytime there's more than one statement within.

Reply:
Huh, I didn't catch that I thought it used following else with the inner most if, good to know, and I really should of added brackets either way it makes it easier to read.